profiles/orcid.py

```diff
@@ -16,37 +16,21 @@
 
 
 class OrcidClient(object):
-    def __init__(self, api_uri: str, token_uri: str, client_id: str, client_secret: str) -> None:
+    # pylint:disable=too-many-arguments
+    def __init__(self, api_uri: str, token_uri: str, client_id: str, client_secret: str,
+                 read_public_access_token: str, webhook_access_token: str) -> None:
         self.api_uri = api_uri
         self.token_uri = token_uri
         self.client_id = client_id
         self.client_secret = client_secret
+        self.read_public_access_token = read_public_access_token
+        self.webhook_access_token = webhook_access_token
```

Comment:
I'd keep these out of the `OrcidClient` now.

Reply:
We are already passing an `OrcidClient` instance to the `webhook_maintainer` and the `_update` view function, if removed, we would then have to also provide the callers access to the `config` somehow. The `OrcidClient` is already being passed the rest of the values for orcid from the config so it seemed to fit to pass it these and provide the same API to the callers `orcid_client.get_access_token()`.

Reply:
We already do that elsewhere (eg `api/oauth2.py` has `orcid: Dict[str, str]`; `OrcidClient` is meant to be like a 3rd-party library so best not to require these two upfront - it's just not be extracted as its incomplete).

Reply:
OK understood. I have now refactored them out of the `OrcidClient`.